src/components/bar/utils/monitors/index.ts

```diff
@@ -12,24 +19,80 @@ import { JSXElement } from 'src/core/types';
  * @returns BarLayout configuration for the specified monitor, falling back to default if not found
  */
 export const getLayoutForMonitor = (monitor: number, layouts: BarLayouts): BarLayout => {
-    const matchingKey = Object.keys(layouts).find((key) => key === monitor.toString());
-    const wildcard = Object.keys(layouts).find((key) => key === '*');
+    const [rootKey, rootLayout] = _getResolveLayoutForMonitor(monitor, layouts);
 
-    if (matchingKey !== undefined) {
-        return layouts[matchingKey];
-    }
+    let left = rootLayout.left;
+    let middle = rootLayout.middle;
+    let right = rootLayout.right;
 
-    if (wildcard) {
-        return layouts[wildcard];
+    let layout = rootLayout;
+    const visited = new Array([rootKey]);
```

Comment:
```suggestion
    const visited = [rootKey];
```
Constructor syntax not needed. The current implementation creates an array of arrays `[["DP-1"]]`. And a single number passed in the `new Array(someVal)` if it ends up being a number creates an array of that size, rather than an array with that element.

The beauty of this awful language lol.